helpers.py

```diff
@@ -0,0 +1,15 @@
+from typing import Any, Optional
+
+
+def get_param_id(obj: Any, key: str) -> Optional[str]:
+    """
+    Returns the params test parameter ID or None
+    """
+    # avoid confusing "TypeError: 'NotSetType' object is not subscriptable" errors
+    if not hasattr(obj, "__getitem__"):
+        return None
+
+    try:
+        return obj[key]
+    except KeyError:
+        return None
```

Comment:
fwiw, a shorter way would be to add the 2nd exception to the same except clause, since they have the same action:
```python

try:
  return obj[key]
except (KeyError, TypeError):
  return None
```

Reply:
+1 I'm in favor of it